src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -1140,7 +1140,7 @@ private static void GetMatchingNamespaceOrTypes(INamespaceOrTypeSymbol container
 
                 foreach (var symbol in members)
                 {
-                    if (symbol.Kind == SymbolKind.Namespace || (symbol.Kind == SymbolKind.NamedType && ((INamedTypeSymbol)symbol).Arity == 0))
+                    if (symbol.Kind == SymbolKind.Namespace || (symbol is INamedTypeSymbol { Arity: 0, Kind: SymbolKind.NamedType }))
```

Comment:
>Kind: SymbolKind.NamedType [](start = 103, length = 26)

This kind test is redundant, but we only know that because of the semantics of the data structure `(symbol.Kind==NamedType) == (symbol is INamedTypeSymbol)`.

Reply:
This falls into a group of observations which make me hesitant to turn this from a refactoring to a diagnostic, but for now it seems like a safe "won't fix".